Make grpc_http1_reverse_bridge return HTTP 200.

hq6 opened the pull request:
gRPC clients expect that request status will be communicated using the gRPC status header, and that HTTP status for responses should always be 200 for properly behaving servers.

This commit updates the grpc_http1_reverse_bridge to follow this behavior.

Testing: Tested manually with grpcurl.

yanavlasov commented:
@hq6 please fix DCO per these instructions https://github.com/envoyproxy/envoy/pull/31047/checks?check_run_id=19010034687

yanavlasov commented on source/extensions/filters/http/grpc_http1_reverse_bridge/filter.cc:
```diff
@@ -179,6 +179,9 @@ Http::FilterHeadersStatus Filter::encodeHeaders(Http::ResponseHeaderMap& headers
     // until then.
     grpc_status_ = grpcStatusFromHeaders(headers);
+
+    // gRPC clients expect that the HTTP status will always be 200.
```
I think you should do it only when it is a gRPC response. For regular HTTP responses the status should not be modified.

Also as this is a user visible change it will require a runtime flag protection.

zuercher replied:
Runtime flags are defined here: https://github.com/envoyproxy/envoy/blob/main/source/common/runtime/runtime_features.cc

zuercher commented:
/wait

hq6 commented:
@yanavlasov @zuercher I have rebased this PR and added runtime protection as requested.

Are there additional changes you'd like to see in this PR?

hq6 commented:
@yanavlasov  @zuercher 
Bumping this since it's been 3 weeks, and I rebased to fix the conflicts.

Are there additional changes you'd like to see in this PR?

adisuissa commented:
/retest

adisuissa commented:
@zuercher PTAL

phlax commented:
@hq6 if you resolve the merge conflict this should be good to land

phlax commented:
@hq6 if you have a reviewer preferably dont force-push

phlax reviewed (approved):
lgtm, thanks @hq6

hq6 commented:
> @hq6 if you resolve the merge conflict this should be good to land

@phlax  I have resolved conflicts again, I do not have write permissions, so I cannot merge it myself.

If possible, could we get this PR stamped and merged before yet another rebase is required?

hq6 commented:
> @hq6 if you have a reviewer preferably dont force-push

Oops I saw this message too late. Sorry I'm used to following a rebase workflow and didn't realize it was causing review dismissal.

phlax commented:
apologies @hq6 i set this to automerge but it flaked and now there is another conflict - would you mind updating again
